superset-frontend/src/components/TableSelector/index.tsx

```diff
@@ -129,179 +175,173 @@ const TableSelector: FunctionComponent<TableSelectorProps> = ({
   tableName,
   tableNameSticky = true,
 }) => {
+  const [currentDbId, setCurrentDbId] = useState<number | undefined>(dbId);
   const [currentSchema, setCurrentSchema] = useState<string | undefined>(
     schema,
   );
-  const [currentTableName, setCurrentTableName] = useState<string | undefined>(
-    tableName,
-  );
-  const [tableLoading, setTableLoading] = useState(false);
-  const [tableOptions, setTableOptions] = useState([]);
-
-  function fetchTables(
-    databaseId?: number,
-    schema?: string,
-    forceRefresh = false,
-    substr = 'undefined',
-  ) {
-    const dbSchema = schema || currentSchema;
-    const actualDbId = databaseId || dbId;
-    if (actualDbId && dbSchema) {
-      const encodedSchema = encodeURIComponent(dbSchema);
-      const encodedSubstr = encodeURIComponent(substr);
-      setTableLoading(true);
-      setTableOptions([]);
+  const [currentTable, setCurrentTable] = useState<TableOption | undefined>();
+  const [refresh, setRefresh] = useState(0);
+
+  const loadTable = useMemo(
+    () => async (dbId: number, schema: string, tableName: string) => {
       const endpoint = encodeURI(
-        `/superset/tables/${actualDbId}/${encodedSchema}/${encodedSubstr}/${!!forceRefresh}/`,
+        `/superset/tables/${dbId}/${schema}/${encodeURIComponent(
+          tableName,
+        )}/true/true`,
       );
```

Comment:
This will force table metadata to be refetched from the db as opposed to the cache. We should probably retain the `forceRefresh` variable and pass it along here, like
```typescript
  const loadTable = useMemo(
    () => async (dbId: number, schema: string, tableName: string, forceRefresh: boolean) => {
      const endpoint = encodeURI(
        `/superset/tables/${dbId}/${schema}/${encodeURIComponent(
          tableName,
        )}/${forceRefresh}/true`,
      );```